open-company/open-company-web
# Special 404 cases

Card: https://trello.com/c/LmyH0IPz

Abstract:
- activity not found https://share.goabstract.com/c672cb21-5cd5-4e04-8971-0a5a52bb8707
- activity removed https://share.goabstract.com/462abebc-d967-4025-90e0-ac8e967f3795

Adds 2 new components called activity-not-found and activity-removed to be used in place of our usual 404 page. The first is used when the user is not logged in and the org/section/activity is not found and prompts to login to see the content, the second is used when the user is logged in but the section/activity required are not found. The special case where the user is actually logged in and the org is not found is handled by showing the usual 404.

There are 6 main cases to test while visiting a made up post url `/{org-slug}/{section-slug}/post/{post-uuid}`:
1. user is not logged in, org not found: show activity not found (prompt to login)
2. user is not logged in, org found, section not found: show activity not found (prompt to login)
3. user is not logged in, org found, section found, activity uuid not found: show activity not found (prompt to login)
4. user is logged in, org not found: show generic 404
5. user is logged in, org found, section not found: show activity removed
6. user is logged in, org found, section found, activity uuid not found: show activity removed

Notice the 4th case is handled separately, it's possible that the user is visiting a deleted section/activity but the org slug wrong probably means the url is wrong.

Tests:
- from desktop, logged out
- visit: /fake-org-slug/fake-section-slug/post/1234-1234-1234
- [x] did you see the new page with a prompt to login? Good
- from desktop, logged out
- visit: /real-org-slug/fake-section-slug/post/1234-1234-1234
- [x] did you see the new page with a prompt to login? Good
- from desktop, logged out
- visit: /real-org-slug/real-section-slug/post/1234-1234-1234
- [x] did you see the new page with a prompt to login? Good
- switch to mobile
- [x] does the page looks good? Good
- back to desktop
- login from here
- from desktop, logged in
- visit: /fake-org-slug/fake-section-slug/post/1234-1234-1234
- [x] did you get the 404 page? Good
- click on Go to digest
- [x] where you NOT redirected back to /fake-org-slug? Good
- from desktop, logged in
- visit: /real-org-slug/fake-section-slug/post/1234-1234-1234
- [x] did you get the new post removed page? Good
- from desktop, logged in
- visit: /real-org-slug/real-section-slug/post/1234-1234-1234
- [x] did you get the new post removed page? Good
- switch to mobile
- [x] does it look good, like in the design (top green icon is wrong in design, check desktop)? Good
- click the Go To Homepage button
- [x] did you get to your current homepage? Good
- now visit your real AP
- switch to grid view
- enter and exit fullscreen for some of your cards
- [x] did it work? Never got the 404, not found, removed activity view? Good
